src/Raven.Client/Documents/Session/Operations/StreamOperation.cs

```diff
@@ -356,12 +358,13 @@ private static void HandleStreamQueryStats(JsonOperationContext context, StreamR
 
             private void CheckIfContextNeedsToBeRenewed()
             {
-                if (_docsCountOnCachedRenewSession <= 16 * 1024)
+                if (_docsCountOnCachedRenewSession <= _maxDocsCountOnCachedRenewSession)
                 {
                     if (_cachedItemsRenew)
                     {
                         _session.Context.CachedProperties = new CachedProperties(_session.Context);
                         ++_docsCountOnCachedRenewSession;
+                        _cachedItemsRenew = false;
```

Comment:
This isn't related to the fix for the NRE but needed to prevent us doing unnecessary work